client/src/components/ScreenShareWindow.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState, useRef } from "react";
+
+function ScreenShareWindow({
+  onStart,
+  onEnd,
+  onFailed,
+}: {
+  onStart: (stream) => void;
+  onEnd: () => void;
+  onFailed: (e) => void;
+}): JSX.Element {
+  useEffect(() => {
+    navigator.mediaDevices
+      .getDisplayMedia()
+      .then((stream) => {
+        onStart(stream);
+        stream.getVideoTracks()[0].onended = () => {
+          onEnd();
+        };
+      })
+      .catch((e) => {
+        onFailed(e);
+      });
+  }, []);
+
+  return <></>;
```

Comment:
It seems like `ScreenShareWindow` is not really a `window` and I don't think this is intuitive.
If getDisplayMedia is the only thing it is doing, I dont think this needs to be React component. More like a function that gets called when user clicks on shareScreenButton.

Reply:
True true I agree. The component kind of diverted from my original idea so I'll try renaming it/refactoring it a bit.